### PR TITLE
New section on previews using GH Actions + update outdated deployment workflows

### LINKED ISSUE #980: Rendering and deploying previews using GitHub
Xref: https://github.com/FluxML/fluxml.github.io/pull/145

Is it possible to use GH pages instead of Netlify to deploy documentation previews with `Franklin`? The [documentation](https://franklinjl.org/workflow/deploy/#previewing_pull_requests) states -

> Now for the final step, you will need to visualize the obtained previews. Since Github only allow you to deploy one branch, you will need an alternative like Netlify. Create an account on [Netlify.com](https://www.netlify.com/), add your repository and chose the gh-preview branch without any additional settings. Set your Netlify website to be {netlify name}.netlify.app.

But, `Documenter.jl` already provides us with the "preview" feature using GH pages. This is due to the fact that Documenter.jl stores both the previews and the actual build in the same branch - `gh-pages`. For example see the [gh-pages branch](https://github.com/FluxML/Flux.jl/tree/gh-pages) branch of `Flux`. The branch has a directory named `previews` to store the previews.

Ideally the following should work, but will it? -
```yaml
- name: Fix URLs for PR preview deployment (pull request previews)
  if: github.event_name == 'pull_request'
  run: |
    echo "PREVIEW_FRANKLIN_WEBSITE_URL=https://fluxml.ai/previews/PR${{ github.event.number }}/" >> $GITHUB_ENV
    echo "PREVIEW_FRANKLIN_PREPATH=previews/PR${{ github.event.number }}" >> $GITHUB_ENV
- name: Deploy (preview)
  if: github.event_name == 'pull_request' && github.repository == github.event.pull_request.head.repo.full_name # if this build is a PR build and the PR is NOT from a fork
  uses: JamesIves/github-pages-deploy-action@releases/v3
  with:
      BRANCH: gh-pages # The branch where the PRs previews are stored
      FOLDER: __site
      GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
      TARGET_FOLDER: "previews/PR${{ github.event.number }}" # The website preview is going to be stored in a subfolder
  - name: Deploy (main)
  if: github.event_name == 'push' && github.ref == 'refs/heads/dev'
  uses: JamesIves/github-pages-deploy-action@releases/v3
  with:
      GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
      BRANCH: gh-pages # Replace here the branch where your website is deployed
      FOLDER: __site
```

### PR DESCRIPTION
**Pull request by Saransh-cpp**
Closes #980 

This PR -
- Adds a new section on how to deploy previews using GH Actions
- Adds a new section on using new GH Pages workflows
- Updates the version and syntax of actions used in the deployment workflow
- Fixes minor styling and grammatical errors

TODO:
- [ ] Add Flux's website as an example once https://github.com/FluxML/fluxml.github.io/pull/145 is merged

### CODE REVIEW

File: docs/workflow/deploy.md [hunk 1]
```diff
@@ -216,11 +238,72 @@ to
...
+```julia
+@def prepath     = "myprepath" # This is possibly "" for your case
+@def website_url = "mywebsite.github.io"
+```
+to
+```julia
+@def prepath     = get(ENV, "PREVIEW_FRANKLIN_PREPATH", "myprepath") # In the third argument put the prepath you normally use
+@def website_url = get(ENV, "PREVIEW_FRANKLIN_WEBSITE_URL", "mywebsite.github.io") # Just put the website name
+```
+
```
Comment (tlienart) [inline docs/workflow/deploy.md:302]: kindly use `+++` syntax for variable definitions so 

```
+++
prepath = get(...)
website_url = get(...)
...
+++
```

`@def` is still correct but will be deprecated

**Review by tlienart — APPROVED** [review]
wonderful, thanks a lot, just a minor request and will wait until you've added the link to your repo then good to merge 🙌

File: docs/workflow/deploy.md (hunk 1 above)
Comment (Saransh-cpp) [reply docs/workflow/deploy.md:310]: Done, thanks!

**Comment by Saransh-cpp** [comment]
Thanks for the review, @tlienart! I will update the example as soon as the port is complete!

**Comment by Saransh-cpp** [comment]
@tlienart, this should be ready now! I have added FluxML's website as an example.

**Review by tlienart — APPROVED** [review]

**Comment by tlienart** [comment]
thanks a lot!

**Pull request merged**